cmake/interrogate_setup_dot_py.py

```diff
@@ -199,6 +198,47 @@ def setup(*args, **kwargs):
         with open(outfile, 'w') as out:
             out.write('\n'.join(result))
 
+        attrs = dict(
+            name=package_name,
+            version=version,
+            package_dir=package_dir,
+            packages=pkgs,
+            scripts=scripts,
+            py_modules=modules,
+            entry_points=entry_points,
+        )
+
+        from distutils.dist import Distribution
+        dist = Distribution(attrs)
+        dist.script_name = 'setup.py'
+
+        # egg_info command
+        from setuptools.command.egg_info import egg_info
+        class mock_egg_info(egg_info):
+            def finalize_options(self):
+                self.egg_name = package_name
+                self.egg_version = version
+                self.egg_base = install_dir
+                self.egg_info = os.path.join(
+                    self.egg_base, '%s-%s.egg-info' % (package_name, version))
```

Comment:
This seems to be pretty fragile to me. The custom function doesn't call the super function. Is this called from anywhere other than two lines below? If not, why is it overriding this method? Couldn't you perform the logic two lines below external of the `egg_info`?